homeassistant/components/bangolufsen/media_player.py

```diff
@@ -0,0 +1,777 @@
+    DOMAIN,
+    ENTITY_ENUM,
+    FALLBACK_SOURCES,
+    HIDDEN_SOURCE_IDS,
+    REPEAT_ENUM,
+    SOURCE_ENUM,
+    VALID_MEDIA_TYPES,
+    WEBSOCKET_NOTIFICATION,
+)
+from .entity import BangOlufsenEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+BANGOLUFSEN_FEATURES = (
+    MediaPlayerEntityFeature.PAUSE
+    | MediaPlayerEntityFeature.SEEK
+    | MediaPlayerEntityFeature.VOLUME_SET
+    | MediaPlayerEntityFeature.VOLUME_MUTE
+    | MediaPlayerEntityFeature.PREVIOUS_TRACK
+    | MediaPlayerEntityFeature.NEXT_TRACK
+    | MediaPlayerEntityFeature.PLAY_MEDIA
+    | MediaPlayerEntityFeature.VOLUME_STEP
+    | MediaPlayerEntityFeature.SELECT_SOURCE
+    | MediaPlayerEntityFeature.STOP
+    | MediaPlayerEntityFeature.CLEAR_PLAYLIST
+    | MediaPlayerEntityFeature.PLAY
+    | MediaPlayerEntityFeature.SHUFFLE_SET
+    | MediaPlayerEntityFeature.BROWSE_MEDIA
+    | MediaPlayerEntityFeature.REPEAT_SET
+    | MediaPlayerEntityFeature.TURN_OFF
+)
+
+
+PARALLEL_UPDATES = 0
+SCAN_INTERVAL = timedelta(minutes=2)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Media Player entity from config entry."""
+    entity = hass.data[DOMAIN][config_entry.unique_id][ENTITY_ENUM.MEDIA_PLAYER]
+    # Add MediaPlayer entity
+    async_add_entities(new_entities=[entity], update_before_add=True)
+
+
+class BangOlufsenMediaPlayer(MediaPlayerEntity, BangOlufsenEntity):
+    """Representation of a media player."""
+
+    _attr_has_entity_name = False
+    _attr_icon = "mdi:speaker-wireless"
+    _attr_supported_features = BANGOLUFSEN_FEATURES
+
+    def __init__(self, entry: ConfigEntry) -> None:
+        """Initialize the media player."""
+        super().__init__(entry)
+
+        self._beolink_jid: str = self.entry.data[CONF_BEOLINK_JID]
+        self._default_volume: int = self.entry.data[CONF_DEFAULT_VOLUME]
+        self._max_volume: int = self.entry.data[CONF_MAX_VOLUME]
+        self._model: str = self.entry.data[CONF_MODEL]
+        self._volume_step: int = self.entry.data[CONF_VOLUME_STEP]
+
+        self._attr_device_class = MediaPlayerDeviceClass.SPEAKER
+        self._attr_device_info = DeviceInfo(
+            configuration_url=f"http://{self._host}/#/",
+            identifiers={(DOMAIN, self._unique_id)},
+            manufacturer="Bang & Olufsen",
+            model=self._model,
+            name=cast(str, self.name),
+        )
+        self._attr_name = self._name
+        self._attr_should_poll = True
+        self._attr_unique_id = self._unique_id
+
+        # Misc. variables.
+        self._audio_sources: dict[str, str] = {}
+        self._beolink_listeners: list[BeolinkListener] = []
+        self._last_update: datetime = datetime(1970, 1, 1, 0, 0, 0, 0)
+        self._media_image: Art = Art()
+        self._queue_settings: PlayQueueSettings = PlayQueueSettings()
+        self._remote_leader: BeolinkLeader | None = None
```

Comment:
This is never set. Please don't add support for potential future features.

Reply:
I've removed the last_update, queue_settings and remote leader.